setup.py

```diff
@@ -51,15 +51,12 @@
 
 
 if sys.platform != 'win32':
-    compile_args = dict( extra_compile_args=['-O2', '-march=native',
-                                             '-Wno-unused-function',
-                                             '-Wno-unneeded-internal-declaration',
-                                             '-Wno-#warnings'],
+    compile_args = dict( extra_compile_args=['-O2', '-march=native'],
                              extra_link_args=['-O2', '-march=native'])
+    libraries = ["m"]
 else:
-    compile_args = dict( extra_compile_args=['-Wunused-function',
-                                              '-Wunneeded-internal-declaration',
-                                              '-Wno-#warnings'])
+    compile_args = dict( extra_compile_args=[])
+    libraries = []
```

Comment:
I had this suspicion (makes sens now, thanks for checking it), and had tried to reimplement the `acosh` function. The problem was that,
  - it worked fine on linux https://travis-ci.org/rth/PyAbel/jobs/189767399
  - but the results were not correct on Windows for some reason https://ci.appveyor.com/project/rth/pyabel/build/1.0.71/job/5hgtn6iic43qukax 

besides I imagine the built-in `acosh` might be better optimized and faster which would be important here.